src/servers/simple.cc

```diff
@@ -107,26 +108,47 @@ ResponseAlloc(
              << tensor_name;
   } else {
     void* allocated_ptr = nullptr;
-    if (memory_type == TRTSERVER_MEMORY_CPU) {
+    if (!use_gpu_memory || (memory_type == TRTSERVER_MEMORY_CPU)) {
       allocated_ptr = malloc(byte_size);
-#ifdef TRTIS_ENABLE_GPU
+      *actual_memory_type = TRTSERVER_MEMORY_CPU;
+      *actual_memory_type_id = 0;
     } else if (use_gpu_memory) {
```

Comment:
This can be unconditional else?

Reply:
Yes. It can. For all other cases CPU will be used except when use_gpu_memory = true and memory_type = TRTSERVER_MEMORY_GPU